tests/jballerina-unit-test/src/test/resources/test-src/error/error_test.bal

```diff
@@ -371,6 +371,31 @@ function testUnionErrorTypeDescriptionInferring() {
     assertEquality(err.detail().toString(), e.detail().toString());
 }
 
+type SampleErrorData record {
+    string message?;
+    error cause?;
+    string info;
+    boolean fatal;
+};
+
+type SampleError error<SampleErrorData>;
+
+function testErrorBindingPattern() {
+    SampleError error(info=info,fatal=fatal) = error SampleError("Sample Error", info = "Detail Info",
+    fatal = true);
+    assertEquality(info, "Detail Info");
+    assertEquality(fatal, true);
+
+    error error(data=transactionData) = error("TransactionError", data={"A":"a", "B":"b"});
+    assertEquality(transactionData.toString(), "{\"A\":\"a\",\"B\":\"b\"}");
+
+    string i;
+    boolean b;
+    error(info=i,fatal=b) = error SampleError("Sample Error", info = "Some Info", fatal = false);
+    assertEquality(i, "Some Info");
+    assertEquality(b, false);
```

Comment:
As `ErrorVariableReferenceTest` is disabled at the moment, we will have to keep this part here.